workshop/content/030_basic_content/075_serverless/_index.md

```diff
@@ -5,7 +5,264 @@ weight: 75
 services: true
 ---
 
-FIS currently does not support disrupting serverless execution in AWS Lambda. It is, however, possible to inject chaos actions by decorating the code executed within AWS Lambda. For a deeper introduction to the topic see:
+FIS currently does not support disrupting serverless execution in [**AWS Lambda**](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html). It is, however, possible to inject chaos actions by decorating the code executed within AWS Lambda. 
+
+In this section we use the open source [**chaos_lambda**](https://github.com/adhorn/aws-lambda-chaos-injection) python library to demonstrate how to
+
+* inject latency into serverless calls,
+* change the response code of the serverless function, and
+* simulate exceptions in code execution.
+
+A similar JS library, [**failure-lambda**](https://github.com/gunnargrosch/failure-lambda) is described in the [**Serverless Chaos workshop**](https://catalog.us-east-1.prod.workshops.aws/workshops/3015a19d-0e07-4493-9781-6c02a7626c65/en-US/serverless/failure-lambda/fault-injection) and the understanding of the priciples should allow the reader to build their own in their preferred language.
+
+## Architecture
+
+In AWS Lambda, serverless functions expose a "handler" function that receives a JSON object and returns a JSON object. We can keep this handler function unchanged by inserting a new wrapper function around the original handler function, e.g. using a [**decorator pattern**](https://en.wikipedia.org/wiki/Decorator_pattern). This wrapper can cause exceptions in lambda function execution before or after customer code is called, can inject latency before or after customer code is called, can modify output thus simulating a failure result, and can even modify input thus triggering failures in the customer function code:
+
+{{< img "serverless.png" "Serverless encapsulation" >}}
+
+In chaos-lambda, to allow injecting failures on-demand, the wrapper function will query an [**SSM Parameter Store parameter **](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) to check whether failures should be injected at all and, if so, what failures.
+
+To inject failures in the context of an FIS experiment, we will use an [**SSM Automation**](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-automation.html) document to change the value of the SSM Parameter Store parameter and turn on different types of failures.
+
+## Experiment idea
+
+In this section we are exploring tooling so we will start without a hypothesis. However, we will provide some learnings and next steps at the end.
```

Comment:
possible re-write: "Since we are exploring the tooling, we will start without defining an hypothesis."

Reply:
Seems unnecessary and a big clunky "However, we will provide some learnings and next steps at the end."